angular2/src/app/app.chat.ts

```diff
@@ -0,0 +1,85 @@
+import {Component, Input, OnInit } from '@angular/core';
+import {ChatEngine} from './chatEngine';
+declare var require: any;
+const typing = require('chat-engine-typing-indicator');
+
+@Component({
+  selector: 'app-chat',
+  templateUrl: './templates/app.chat.html'
+})
+export class AppChatComponent implements OnInit {
+  private ce: any;
+  @Input() chat: any;
+  @Input() index: number;
+  messages: any[] = [];
+  message: string;
+  mysearch: string = '';
+  users: any[] = [];
+
+  constructor(private chatEngine: ChatEngine) {
+    this.ce = chatEngine;
+  }
+
+  ngOnInit() {
+    this.chat.plugin(typing({ timeout: 5000 }));
+    this.chat.on('message', (payload) => {
+      // if the last message was sent from the same user
+      payload.sameUser = this.messages.length > 0 && payload.sender.uuid === this.messages[this.messages.length - 1].sender.uuid;
+
+      // if this message was sent by this client
+      payload.isSelf = payload.sender.uuid === this.ce.me.uuid;
```

Comment:
You can do ```payload.sender.name == 'Me'```

Reply:
ready, in this commit is done the change: https://github.com/pubnub/chat-engine-examples/pull/12/commits/36ec44b0aef6670053efa26f27ecc48ee2f469ac